src/core/Events.js

```diff
@@ -159,6 +159,9 @@ L.Evented = L.Class.extend({
 					}
 				}
 			}
+			if (listeners.length === 0) {

```

Comment:
I'd move this in the previous `if`, because it seems to me that we only want to check `length` if `listeners` is not `undefined`/`null`.